### ISSUE TITLE
Exploding gradients in training BERT from scratch

### ISSUE THREAD

**Issue by fajri91** [google-research/bert#1016]
Hi, I am training (base) BERT with 4 GPU V100 (each 16GB) with this training configuration:
_Data is only 200million words_
--train_batch_size=8
--max_seq_length=512
--max_predictions_per_seq=75
--num_train_steps=7000000
--num_warmup_steps=70000
--learning_rate=1e-4
--vocab_size=70K

![Screenshot from 2020-03-02 07-31-32](https://user-images.githubusercontent.com/12390478/75633842-c03a1500-5c5c-11ea-9a37-00d047ae017e.png)

Shouldn't this happen as the training already include gradient clipping? `(grads, _) = tf.clip_by_global_norm(grads, clip_norm=1.0)`  
Or could it be because my vocab size is too big for my current data? I notice BERT vocab is only 30K.

**Comment by dhruvsakalley** [comment]
My first observation is your batch size is very low, do remember that google bert code is not optimized to run on multiple GPUs, I highly recommend using this instead: https://github.com/NVIDIA/DeepLearningExamples/tree/master/PyTorch/LanguageModeling/BERT
this branch will help if you have scaling happening on a single machine with multiple GPUs

**Comment by fajri91** [comment]
Hi @dhruvsakalley thanks for your reply.
You are right, its not for multi-gpu training. I 've checked my training, even with 1 GPU and batch_size 8, it works properly.

Thank you for your suggestion. Really help!
